### ISSUE TITLE
Pachyderm needs a minio option for custom object store deployments

### ISSUE THREAD

**Issue by Nick-Harvey** [pachyderm/pachyderm#3711]
**What is the goal / desired outcome?**
Many users are looking to leverage minio as their object store. As it stands right now, Pachyderm expects a region to be defined as part of the object store deployment and minio doesn't require that to be defined in their default deployments and therefore causes `pachctl deploy custom` to fail. We should have a defined `pachctl deploy --persistent-disk minio ..` which doesn't expect a region by default.  

**If there is a way to accomplish this today via workaround, what does that require?**
Users have to configure their minio setup to have a region defined, and then users should be able to deploy via `pachctl deploy custom --persistent-disk aws ...` but this isn't well-known or extensively tested.

**(Optional) What is your proposal for a feature to solve this?**
Create a `pachctl deploy custom --persistent-disk minio` option 

**Environment?**:
- Kubernetes version (use `kubectl version`):
- Pachyderm CLI version (use `pachctl version`):
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
- OS (e.g. from /etc/os-release):
- Others:

**Comment by jdoliner** [comment]
I don't think this issue makes sense, afaik minio does not provide persistent disks, it only provides object storage. Users have to get the persistent disk from somewhere else.

**Comment by Nick-Harvey** [comment]
I got the `--persistent-disk` part wrong. Meant to say `--object-store minio ...`

**Comment by Nick-Harvey** [comment]
Looks like we have this implemented at least to some degree: https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/server/server.go 

```
const (
    MinioBackendEnvVar     = "MINIO"
    AmazonBackendEnvVar    = "AMAZON"
    GoogleBackendEnvVar    = "GOOGLE"
    MicrosoftBackendEnvVar = "MICROSOFT"
    LocalBackendEnvVar     = "LOCAL"
)
``` 

not sure how deep this rabbit hole goes though :)

**Comment by gabrielgrant** [comment]
I think it _may_ just be that we should have a default region of `us-east-1` if one isn't specified, as this is the default/hard-coded value in v4 (our default version) of the S3 API: https://github.com/pachyderm/pachyderm/blob/fcfd38ef20db2402df3c969c0ebd1a82f680c824/src/server/vendor/github.com/minio/minio-go/api-put-bucket.go#L193
